source/custom/nixnet_service.custom.cpp

```diff
@@ -1218,6 +1218,22 @@ void convert_to_grpc(const void* input, nixnet_grpc::FrameBuffer* output, u32 fr
     }
   }
 }
+
+u32 determine_size_from_request(u32 number_of_signals, u32 number_of_values)
+{
+  return (number_of_signals * number_of_values * sizeof(f64));
+}
+
+u32 determine_size_of_output_buffer(u32 size_of_value_buffer, u32 size_of_timestamp_buffer)
+{
+  if(size_of_value_buffer == 0U) {
+    return size_of_timestamp_buffer;
+  }
+  if(size_of_timestamp_buffer == 0U) {
+    return size_of_value_buffer;
+  }
+  return std::min(size_of_timestamp_buffer, size_of_value_buffer);
+}
```

Comment:
For ReadSignalXY API, if both sizeofvaluebuffer and sizeoftimestampbuffer are non zero values then, minimum has to be taken for both valuebuffer and timestampbuffer as per the documentation.